src/lib/useRollerPoller.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from 'react';
+import { useRollerStore } from 'store/rollerStore';
+import { useTimerStore } from 'store/timerStore';
+import { ONE_SECOND, TEN_SECONDS } from './constants';
+import useRoller from './useRoller';
+import { getTimeToNextBatch } from './utils/roller';
+
+export function useRollerPoller() {
+  const { nextBatchTime, setNextBatchTime, setNextRoll } = useTimerStore();
+  const { point } = useRollerStore();
+  const { api, getInvites, getPendingTransactions } = useRoller();
+  const [time, setTime] = useState(ONE_SECOND);
+
+  useEffect(() => {
+    const interval = setInterval(async () => {
+      const nextRoll = getTimeToNextBatch(nextBatchTime, new Date().getTime());
+      setNextRoll(nextRoll);
+
+      if (nextBatchTime - ONE_SECOND <= new Date().getTime()) {
```

Comment:
I'm still unsure about this logic because it ends up firing multiple times. We could lock this down so it only fires once in that timeframe, but may be worth putting off.